ros/src/waypoint_updater/waypoint_updater.py

```diff
@@ -21,36 +25,164 @@
 TODO (for Yousuf and Aaron): Stopline location for each traffic light.
 '''
 
-LOOKAHEAD_WPS = 200 # Number of waypoints we will publish. You can change this number
+# Set Global Variables to use
+
+LOOKAHEAD_WPS = 50 # Number of waypoints we will publish. You can change this number
+LOOKAHEAD_WPS_FOR_STOPPING = 50
+MAX_SPEED = 11.111 # A predefined speed of 11.1111 m/s from waypoint to waypoint
+STOPLINE_BUFFER = 2. # Car stops 2m infront of the stop line.
+DISTANCE_TOO_CLOSE_FOR_STOPPING = 10.# If the car is at full speed and less then this distance ahead of the stop line
 
 
 class WaypointUpdater(object):
     def __init__(self):
+
+        self.waypoint_tree = None
+        self.pose = None
+        self.latest_stop_line_idx = -1
+        self.stopping_start_position_found = False
+        self.total_stop_lenght = 0
+        self.curr_vel = None
+
+        # Intiate the Ros Node
         rospy.init_node('waypoint_updater')
 
         rospy.Subscriber('/current_pose', PoseStamped, self.pose_cb)
         rospy.Subscriber('/base_waypoints', Lane, self.waypoints_cb)
+        rospy.Subscriber('/current_velocity', TwistStamped, self.car_curr_vel_cb)
 
         # TODO: Add a subscriber for /traffic_waypoint and /obstacle_waypoint below
 
-
-        self.final_waypoints_pub = rospy.Publisher('final_waypoints', Lane, queue_size=1)
+        self.sub_tl_idx = rospy.Subscriber('/traffic_waypoints', Int32, self.traffic_cb)
```

Comment:
Do we want to implement this right from the outset? Until we work on the traffic light detection part, wouldn't it be better to simply ignore traffic lights?